core/templates/domain/improvements/HighBounceRateTaskObjectFactory.ts

```diff
@@ -0,0 +1,139 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Domain object for a high bounce-rate improvements task.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import { ExplorationStats } from
+  'domain/statistics/ExplorationStatsObjectFactory';
+import { ITaskEntryBackendDict, TaskEntry } from
+  'domain/improvements/TaskEntryObjectFactory';
+import { ImprovementsConstants } from
+  'domain/improvements/improvements.constants';
+
+export class HighBounceRateTask extends TaskEntry {
+  constructor(
+      expId: string,
+      expVersion: number,
+      stateName: string,
+      issueDescription: string = null,
+      taskStatus: string = ImprovementsConstants.TASK_STATUS_TYPE_OBSOLETE) {
+    super(
+      ImprovementsConstants.TASK_ENTITY_TYPE_EXPLORATION, expId, expVersion,
+      ImprovementsConstants.TASK_TYPE_HIGH_BOUNCE_RATE,
+      ImprovementsConstants.TASK_TARGET_TYPE_STATE,
+      stateName, null, null, issueDescription, taskStatus);
+  }
+
+  public resolve(): void {
+    this.markAsResolved();
+  }
+
+  public refreshStatus(expStats: ExplorationStats): void {
+    if (expStats.expId !== this.entityId ||
+        expStats.expVersion !== this.entityVersion) {
+      throw new Error(
+        'Expected stats for exploration ' + (
+          'id="' + this.entityId + '" v' + this.entityVersion) +
+        ' but given stats are for exploration ' + (
+          'id="' + expStats.expId + '" v' + expStats.expVersion));
+    }
+    const expStarts = expStats.numStarts;
+    const bounceRate = expStats.getBounceRate(this.targetId);
+    if (this.meetsCreationConditions(expStarts, bounceRate)) {
+      this.markAsOpen();
+      this.generateIssueDescription(bounceRate);
+    } else if (this.meetsObsoletionConditions(expStarts, bounceRate)) {
+      this.markAsObsolete();
+    }
+  }
+
+  private meetsCreationConditions(
+      expStarts: number, bounceRate: number): boolean {
+    return (
+      this.isObsolete() &&
+      expStarts >= ImprovementsConstants.HIGH_BOUNCE_RATE_MIN_EXP_STARTS &&
+      bounceRate >= ImprovementsConstants.HIGH_BOUNCE_RATE_THRESHOLD_HIGH);
+  }
+
+  private meetsObsoletionConditions(
```

Comment:
Apparently "obsoletion" is "a rare word for obsolescence". Maybe use the latter instead? https://www.collinsdictionary.com/us/dictionary/english/obsoletion

(I don't care too much either way though. It's your call :P)

Reply:
OK, I'll stick with obsoletion because it fits more nicely with completion and because I'd like the symmetry between them to be obvious through name.